helm/templates/statefulset.yaml

```diff
@@ -23,6 +23,48 @@ spec:
         {{- end }}
     spec:
       terminationGracePeriodSeconds: {{ $.Values.deployment.terminationGracePeriodSeconds }}
+      securityContext:
+        fsGroup: 1111
+      initContainers:
+        - name: memlock-setup
+          image: busybox:latest
+          command: ["sh", "-c"]
+          args:
+            - |
+              REQUIRED_MEMLOCK_BYTES=8589934592 # 8GB
+              # Expected: 'unlimited' or a number in bytes.
+              CURRENT_MEMLOCK_VALUE=$(cat /proc/self/limits | grep -F 'Max locked memory' | awk '{print $4}')
+              echo "Current soft memlock limit: $CURRENT_MEMLOCK_VALUE"
+
+              if [ "$CURRENT_MEMLOCK_VALUE" = "unlimited" ]; then
+                echo "Current memlock limit is 'unlimited'. No change needed."
+              else
+                if [ "$CURRENT_MEMLOCK_VALUE" -lt "$REQUIRED_MEMLOCK_BYTES" ]; then
+                  echo "Current limit ($CURRENT_MEMLOCK_VALUE bytes) is below required 8GB. Attempting to set new limit..."
+                  if ulimit -l "$REQUIRED_MEMLOCK_BYTES"; then
+                    echo "Memlock ulimit successfully set to $REQUIRED_MEMLOCK_BYTES bytes."
+                  else
+                    echo "Failed to set memlock ulimit. Check capabilities or node configuration." 1>&2
+                    exit 1
+                  fi
+                else
+                  echo "Current memlock limit ($CURRENT_MEMLOCK_VALUE bytes) is already sufficient (>= 8GB). No change needed."
+                fi
+              fi
+
+              # Verify the final memlock limit
+              echo "Final memlock limit after initContainer operation:"
+              cat /proc/self/limits | grep -F 'Max locked memory'
+          securityContext:
+            # Grant CAP_IPC_LOCK to allow setting the memlock ulimit.
+            # Grant CAP_SYS_RESOURCE as it's often needed for `ulimit` operations.
+            capabilities:
+              add:
```

Comment:
You can additionally add:
```
drop:
  - ALL
```

This will drop all capabilities that were inherited.

Reply:
Seems to work